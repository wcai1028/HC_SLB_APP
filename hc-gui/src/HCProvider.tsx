import React from 'react'
import { Store } from 'redux'
import { A10Provider } from 'a10-gui-framework'

export interface IHCPRoviderProps {
  children: JSX.Element
  store?: Store<any>
  initState?: IObject
}

import appReducers from './redux/reducers'
import config from './settings/config'
const middlewares: any[] = []

const HCProvider = (props: IHCPRoviderProps) => {
  const { children, store, initState } = props
  const options: any = {
    initState,
    reducers: appReducers,
    middlewares,
    CONFIG: config,
    children,
  }
  if (store) {
    options.store = store
  }
  return <A10Provider {...options} />
}
export default HCProvider
