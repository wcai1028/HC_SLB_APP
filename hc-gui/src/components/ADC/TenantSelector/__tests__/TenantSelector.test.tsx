import * as React from 'react'
import * as ReactTestRenderer from 'react-test-renderer'
import HCProvider from '../../../../HCProvider'
import { BrowserRouter, Route } from 'react-router-dom'
import { TenantSelector } from '../index'

const renderTestComponent = props => {
  return ReactTestRenderer.create(
    <HCProvider>
      <BrowserRouter>
        <TenantSelector {...props} />
      </BrowserRouter>
    </HCProvider>,
  )
}

test('should render properly with current option', () => {
  const onCurrentChange = jest.fn()
  const props = { options: ['tenant1', 'tenant2', 'root'], currentOption: 'tenant2', onCurrentChange }
  const testComponent = renderTestComponent(props)
  expect(testComponent.toJSON()).toMatchSnapshot()
  expect(onCurrentChange.mock.calls.length).toEqual(1)
})

test('should render properly with default option', () => {
  const onCurrentChange = jest.fn()
  const props = { options: ['tenant1', 'tenant2', 'root'], currentOption: null, onCurrentChange, defaultOption: 'root' }
  const testComponent = renderTestComponent(props)
  expect(testComponent.toJSON()).toMatchSnapshot()
})

test('function componentDidUpdate', () => {
  const onCurrentChange = jest.fn()
  const props = { options: ['tenant1', 'tenant2', 'root'], currentOption: 'tenant2', onCurrentChange }
  const testComponent = renderTestComponent(props)
  const selector = testComponent.root.find((node: any) => (node.type as any).name === 'TenantSelector')
  selector.instance.componentDidUpdate({ currentOption: 'tenant2' })
  expect(selector.instance.state.currentIndex).toBe(1)
})

test('function onControlDropMenu', () => {
  const onCurrentChange = jest.fn()
  const props = { options: ['tenant1', 'tenant2', 'root'], currentOption: 'tenant2', onCurrentChange }
  const testComponent = renderTestComponent(props)
  const selector = testComponent.root.find((node: any) => (node.type as any).name === 'TenantSelector')
  selector.instance.onControlDropMenu(true)
  expect(selector.instance.state.showDropMenu).toBeTruthy()
})

test('function onClickOption', () => {
  const onCurrentChange = jest.fn()
  const props = { options: ['tenant1', 'tenant2', 'root'], currentOption: 'tenant2', onCurrentChange }
  const testComponent = renderTestComponent(props)
  const selector = testComponent.root.find((node: any) => (node.type as any).name === 'TenantSelector')
  selector.instance.onClickOption('tenant1', 0)
  expect(selector.instance.state.currentIndex).toBe(0)
  expect(onCurrentChange.mock.calls.length).toEqual(2)
})
