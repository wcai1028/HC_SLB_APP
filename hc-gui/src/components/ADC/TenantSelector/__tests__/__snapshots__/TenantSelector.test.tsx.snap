// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly with current option 1`] = `
<div
  className="tenantSelector"
>
  <div
    className="currentBlock"
    onClick={[Function]}
  >
    <div
      className="currentIcon"
    >
      <span>
        T
      </span>
    </div>
    <span
      className="currentText"
    >
      tenant2
    </span>
    <i
      className="anticon anticon-down currentArrow"
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="down"
        fill="currentColor"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
        />
      </svg>
    </i>
  </div>
</div>
`;

exports[`should render properly with default option 1`] = `
<div
  className="tenantSelector"
>
  <div
    className="currentBlock"
    onClick={[Function]}
  >
    <div
      className="currentIcon"
    >
      <span>
        R
      </span>
    </div>
    <span
      className="currentText"
    >
      root
    </span>
    <i
      className="anticon anticon-down currentArrow"
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="down"
        fill="currentColor"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
        />
      </svg>
    </i>
  </div>
</div>
`;
